### PR TITLE
Replaces bad Destroy() calls with qdel()

### PR DESCRIPTION
**Pull request by mrmanlikesbt**
## About The Pull Request

Calling `Destroy()` is not how you delete things, come on guys.

## Testing Photographs and Procedure

image

## Changelog
:cl:
code: Replaced some bad Destroy() calls with the proper qdel() calls.
/:cl:

### CODE REVIEW

File: code/modules/vehicles/mecha/equipment/tools/other_tools.dm [hunk 1]
```diff
@@ -513,8 +513,7 @@
 		name = existing_equip.name
 		icon = existing_equip.icon
 		icon_state = existing_equip.icon_state
-		existing_equip.detach()
-		existing_equip.Destroy()
```
Comment (mrmanlikesbt) [inline code/modules/vehicles/mecha/equipment/tools/other_tools.dm:517]: `/obj/item/mecha_parts/mecha_equipment/Destroy()` calls `detach()`

**Comment by PowerfulBacon** [comment]
We could mark Destroy with private proc and then make qdel call it with UNLINT to let it pass. Then itd error out anywhere its used poorly

**Comment by mrmanlikesbt** [comment]
> We could mark Destroy with private proc and then make qdel call it with UNLINT to let it pass. Then itd error out anywhere its used poorly

yea, you're right

**Comment by PowerfulBacon** [comment]
Oh yea it has to be protected for the subtype which makes it not that useful. I guess it helps with global proc calls at least

File: code/modules/surgery/advanced/lobotomy.dm [hunk 2]
```diff
@@ -59,8 +59,8 @@
 
 	target.cure_all_traumas(TRAUMA_RESILIENCE_LOBOTOMY)
 	if(target.ai_controller)
-		target.ai_controller.Destroy()	//we just cut a piece of an already simple mind, there is no turning back.
-	if(target.mind && target.mind.has_antag_datum(/datum/antagonist/brainwashed))
+		qdel(target.ai_controller) //we just cut a piece of an already simple mind, there is no turning back.
```
Comment (Absolucy) [inline code/modules/surgery/advanced/lobotomy.dm]:
```suggestion
		QDEL_NULL(target.ai_controller) //we just cut a piece of an already simple mind, there is no turning back.
```
otherwise won't this hard delete?

**Review by Absolucy — CHANGES_REQUESTED** [review]

File: code/modules/surgery/advanced/lobotomy.dm (hunk 2 above)
Comment (mrmanlikesbt) [reply code/modules/surgery/advanced/lobotomy.dm]: no, since `/datum/ai_controller/Destroy()` calls `UnpossessPawn()` which clears the pawn's `ai_controller` ref.

but doing `target.ai_controller.UnpossessPawn(destroy = TRUE)` would probably be better in this case

Comment (Absolucy) [reply code/modules/surgery/advanced/lobotomy.dm]: oh, fair

File: code/modules/shuttle/shuttle_creation/shuttle_creator.dm [hunk 3]
```diff
@@ -211,16 +211,16 @@ GLOBAL_LIST_EMPTY(custom_shuttle_machines)		//Machines that require updating (He
...
 		return FALSE
 
 	var/datum/map_template/shuttle/new_shuttle = new /datum/map_template/shuttle()
 
-	var/obj/docking_port/stationary/stationary_port = new /obj/docking_port/stationary(get_turf(target))
+	var/obj/docking_port/stationary/stationary_port = new(get_turf(target))
 	stationary_port.delete_after = TRUE
 	stationary_port.name = "[recorded_shuttle_area.name] Custom Shuttle construction site"
-	var/obj/docking_port/mobile/port = new /obj/docking_port/mobile(get_turf(target))
+	var/obj/docking_port/mobile/port = new(get_turf(target))
```
Comment (DrDuckedGoose) [inline code/modules/shuttle/shuttle_creation/shuttle_creator.dm]: Possible micro optimization with the turf target here? I doubt it matters, not like it's called often anyway

**Review by DrDuckedGoose — APPROVED** [review]

File: code/modules/shuttle/shuttle_creation/shuttle_creator.dm (hunk 3 above)
Comment (mrmanlikesbt) [reply code/modules/shuttle/shuttle_creation/shuttle_creator.dm]: Sure, I'll do that.
